### PR TITLE
[skip ci] dashboard: remove "certificate is valid for" error

### PR DESCRIPTION
**Pull request by guits**
When deploying dashboard with ssl certificates generated by
ceph-ansible, we enforce the CN to 'ceph-dashboard' which can makes
application such alertmanager complain like following:

`err="Post https://mgr0:8443/api/prometheus_receiver: x509: certificate is valid for ceph-dashboard, not mgr0" context_err="context deadline exceeded"`

The idea here is to add alternative names matching all mgr/mon instances
in the certificate so this error won't appear in logs.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1978869

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 1]
```diff
@@ -8,6 +8,16 @@
   set_fact:
     ceph_cmd: "{{ hostvars[groups[mon_group_name][0]]['container_binary'] + ' run --interactive --net=host --rm -v /etc/ceph:/etc/ceph:z --entrypoint=ceph ' + ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment | bool else 'ceph' }}"
 
+- name: get current mgr backend - ipv4
+  set_fact:
+    dashboard_server_addr: "{{ hostvars[item]['ansible_facts']['all_ipv4_addresses'] | ips_in_ranges(dashboard_network.split(',')) | first }}"
```
Comment (dsavineau) [inline roles/ceph-dashboard/tasks/configure_dashboard.yml:13]:
```suggestion
    dashboard_server_addr: "{{ ansible_facts['all_ipv4_addresses'] | ips_in_ranges(dashboard_network.split(',')) | first }}"
```

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 2]
```diff
@@ -8,6 +8,16 @@
...
+  when: ip_version == 'ipv4'
+
+- name: get current mgr backend - ipv6
+  set_fact:
+    dashboard_server_addr: "{{ hostvars[item]['ansible_facts']['all_ipv6_addresses'] | ips_in_ranges(dashboard_network.split(',')) | last }}"
```
Comment (dsavineau) [inline roles/ceph-dashboard/tasks/configure_dashboard.yml:18]: ditto

**Review by dsavineau — CHANGES_REQUESTED** [review]

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Review by dsavineau — APPROVED** [review]

**Pull request merged**